src/main/java/voidcat/storage/Storage.java

```diff
@@ -0,0 +1,84 @@
+package voidcat.storage;
+
+import voidcat.task.*;
+import voidcat.exception.VoidException;
+import voidcat.ui.Ui;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class Storage {
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws IOException, VoidException, SecurityException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        File file = new File(filePath);
+        if (file.exists()) {
+            try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    Task task = parseTask(line);
+                    if (task != null) {
+                        tasks.add(task);
+                    }
+                }
+                if (tasks.isEmpty()) {
+                    Ui.showMessageAndLines("No saved tasks found yet! voidcat.task.Task list is empty.\n\tStart adding tasks and track them!");
```

Comment:
Your `Storage` class is handling file operations really well! To make it even better, consider avoiding direct interactions with the `Ui` class. Instead, throw exceptions for any errors and handle them in the `Void` class. This way, your `Storage` class stays focused on its main job i.e. managing files.